pubtools/_impl/pluggy.py

```diff
@@ -94,4 +94,22 @@ def task_context():
         pm.hook.task_stop(failed=failed)
 
 
+@hookspec(firstresult=True)
+def get_cert_key_paths_plugin(server_url):
```

Comment:
Just realized the name here should change.

I don't think it makes sense to name this with `_plugin`.  If it's meant to indicate that this is supposed to be provided by a plugin: the same applies to all hooks, so it would seem that either all of them should be named as `_plugin` or none of them should be... and right now it's none of them.

Can you please remove the `_plugin` for consistency with the other hooks?

Reply:
'get_cert_key_paths' is already existing function defined and used throughout pub. I created a small wrapper function for pluggy and gave it _plugin suffix. I think you haven't noticed my accompanying pub code change: https://code.engineering.redhat.com/gerrit/c/rcm-pub/+/261357

I support it's possible to add hookimpl decorator to the original function, but I haven't tested if it would still work if used normally, and I'm not sure if it's a good decision, design-wise. What do you think?

Reply:
Yes, I've read that change but I don't see much relevance here - it doesn't follow, that the presence of a function named X in Pub means that we shouldn't have a hook named X.  Why would that be the case?  And even if it was, I think it's the Pub side which should compromise - the naming here should make sense on its own merits, we shouldn't let a legacy codebase drag it down and start putting illogical names on things because it happens to make integration with the old code slightly easier.

> I support it's possible to add hookimpl decorator to the original function

It should be safe.  If you look at what the decorator does, it won't affect regular calls to the function.

The other solution which comes to mind: put the (existing function) and (wrapper function + call to pm.register) in different modules.  There's no need for them to be in the same module, therefore no need for them to have different names.

This might be a good idea for code organization too, e.g. it seems nice being able to find all the hookimpl in Pub in one place rather than scattered around.

```python
# maybe in a module named e.g. 'hooks' which has all the top-level hookimpls for integrating (pub <=> pubtools) ?
from pubd[...] import get_cert_key_paths as pub_get_cert_key_paths

@hookimpl
def get_cert_key_paths(...):
    return pub_get_cert_key_paths(...)
```

I think either of these would be better than compromising on the hook naming here.

Reply:
Changed the name.